### PR TITLE
TEST/JENKINS: added build by clang

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -227,6 +227,26 @@ build_icc() {
...
+build_clang() {
+	echo 1..1 > build_clang.tap
+    if which clang > /dev/null 2>&1
+	then
+		echo "==== Build with clang compiler ===="
+		../contrib/configure-devel --prefix=$ucx_inst CC=clang CXX=clang++
+		$MAKE clean
+		$MAKE
+		$MAKE distclean
+		module unload intel/ics
```
Comment (miked-mellanox) [inline contrib/test_jenkins.sh]: why intel/ics is in use here?

Comment (hoopoepg) [reply contrib/test_jenkins.sh]: artefact (copy-paste issue)
removed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4564/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4565/ for details (Mellanox internal link).

**Comment by miked-mellanox** [comment]
@yosefe

File: contrib/test_jenkins.sh [hunk 2]
```diff
@@ -227,6 +227,25 @@ build_icc() {
 	fi
 }
 
+#
+# Build with clang compiler
+#
+build_clang() {
+	echo 1..1 > build_clang.tap
+    if which clang > /dev/null 2>&1
```
Comment (yosefe) [inline contrib/test_jenkins.sh]: indentation

File: contrib/test_jenkins.sh [hunk 3]
```diff
@@ -527,6 +546,7 @@ run_tests() {
 	export UCX_ERROR_MAIL_FOOTER=$JOB_URL/$BUILD_NUMBER/console
 
 	do_distributed_task 0 4 build_icc
+	do_distributed_task 0 4 build_clang
```
Comment (yosefe) [inline contrib/test_jenkins.sh]: pls use "3 4" instead of "0 4" to distribute the tasks more equally

File: contrib/test_jenkins.sh (hunk 2 above)
Comment (hoopoepg) [reply contrib/test_jenkins.sh]: fixed

File: contrib/test_jenkins.sh (hunk 3 above)
Comment (hoopoepg) [reply contrib/test_jenkins.sh]: updated

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4568/ for details (Mellanox internal link).

**Pull request merged**